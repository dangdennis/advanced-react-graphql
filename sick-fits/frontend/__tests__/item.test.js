import ItemComponent from '../components/Item';
import { shallow, mount } from 'enzyme';
import Item from '../components/styles/ItemStyles';

const fakeItem = {
  id: 'ABC',
  title: 'cool item',
  description: 'cool desc',
  price: 5000,
  image: 'test.jpg',
  largeImage: 'largeImage.jpg'
};

describe('<Item/>', () => {
  it('renders and matches the snapshot', () => {
    const wrapper = shallow(<ItemComponent item={fakeItem} />);
    expect(wrapper).toMatchSnapshot();
  })
  // it('renders the price tag and title', () => {
  //   const wrapper = shallow(<ItemComponent item={fakeItem} />);
  //   const PriceTag = wrapper.find('PriceTag');
  //   expect(PriceTag.children().text()).toBe('$50');
  //   expect(wrapper.find('Title a').text()).toBe(fakeItem.title);
  // });

  // it('renders the image properly', () => {
  //   const wrapper = shallow(<ItemComponent item={fakeItem} />);
  //   const img = wrapper.find('img');
  //   expect(img.props().src).toBe(fakeItem.image);
  //   expect(img.props().alt).toBe(fakeItem.title);
  // });

  // it('renders buttons', () => {
  //   const wrapper = shallow(<ItemComponent item={fakeItem} />);
  //   const buttonList = wrapper.find('.buttonList');
  //   expect(buttonList.children()).toHaveLength(3);
  //   expect(buttonList.find('Link')).toHaveLength(1);
  //   expect(buttonList.find('AddToCart').exists()).toBe(true);
  //   expect(buttonList.find('DeleteItem').exists()).toBe(true);
  // });
});
