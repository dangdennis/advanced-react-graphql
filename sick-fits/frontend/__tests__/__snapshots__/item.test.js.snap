// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item/> renders and matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Item
    item={
      Object {
        "description": "cool desc",
        "id": "ABC",
        "image": "test.jpg",
        "largeImage": "largeImage.jpg",
        "price": 5000,
        "title": "cool item",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <img
          alt="cool item"
          src="test.jpg"
        />,
        <Title>
          <Link
            href={
              Object {
                "pathname": "/item",
                "query": Object {
                  "id": "ABC",
                },
              }
            }
          >
            <a>
              cool item
            </a>
          </Link>
        </Title>,
        <PriceTag>
          $50
        </PriceTag>,
        <p>
          cool desc
        </p>,
        <div
          className="buttonList"
        >
          <Link
            href={
              Object {
                "pathname": "update",
                "query": Object {
                  "id": "ABC",
                },
              }
            }
          >
            <a>
              Edit ✏️
            </a>
          </Link>
          <AddToCart
            id="ABC"
          />
          <DeleteItem
            id="ABC"
          >
            Delete This Item
          </DeleteItem>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "cool item",
          "src": "test.jpg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Link
            href={
              Object {
                "pathname": "/item",
                "query": Object {
                  "id": "ABC",
                },
              }
            }
          >
            <a>
              cool item
            </a>
          </Link>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <a>
              cool item
            </a>,
            "href": Object {
              "pathname": "/item",
              "query": Object {
                "id": "ABC",
              },
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "cool item",
            },
            "ref": null,
            "rendered": "cool item",
            "type": "a",
          },
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "$50",
        },
        "ref": null,
        "rendered": "$50",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "cool desc",
        },
        "ref": null,
        "rendered": "cool desc",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Link
              href={
                Object {
                  "pathname": "update",
                  "query": Object {
                    "id": "ABC",
                  },
                }
              }
            >
              <a>
                Edit ✏️
              </a>
            </Link>,
            <AddToCart
              id="ABC"
            />,
            <DeleteItem
              id="ABC"
            >
              Delete This Item
            </DeleteItem>,
          ],
          "className": "buttonList",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <a>
                Edit ✏️
              </a>,
              "href": Object {
                "pathname": "update",
                "query": Object {
                  "id": "ABC",
                },
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Edit ✏️",
              },
              "ref": null,
              "rendered": "Edit ✏️",
              "type": "a",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "id": "ABC",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Delete This Item",
              "id": "ABC",
            },
            "ref": null,
            "rendered": "Delete This Item",
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <img
            alt="cool item"
            src="test.jpg"
          />,
          <Title>
            <Link
              href={
                Object {
                  "pathname": "/item",
                  "query": Object {
                    "id": "ABC",
                  },
                }
              }
            >
              <a>
                cool item
              </a>
            </Link>
          </Title>,
          <PriceTag>
            $50
          </PriceTag>,
          <p>
            cool desc
          </p>,
          <div
            className="buttonList"
          >
            <Link
              href={
                Object {
                  "pathname": "update",
                  "query": Object {
                    "id": "ABC",
                  },
                }
              }
            >
              <a>
                Edit ✏️
              </a>
            </Link>
            <AddToCart
              id="ABC"
            />
            <DeleteItem
              id="ABC"
            >
              Delete This Item
            </DeleteItem>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "cool item",
            "src": "test.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Link
              href={
                Object {
                  "pathname": "/item",
                  "query": Object {
                    "id": "ABC",
                  },
                }
              }
            >
              <a>
                cool item
              </a>
            </Link>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <a>
                cool item
              </a>,
              "href": Object {
                "pathname": "/item",
                "query": Object {
                  "id": "ABC",
                },
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "cool item",
              },
              "ref": null,
              "rendered": "cool item",
              "type": "a",
            },
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "$50",
          },
          "ref": null,
          "rendered": "$50",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "cool desc",
          },
          "ref": null,
          "rendered": "cool desc",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Link
                href={
                  Object {
                    "pathname": "update",
                    "query": Object {
                      "id": "ABC",
                    },
                  }
                }
              >
                <a>
                  Edit ✏️
                </a>
              </Link>,
              <AddToCart
                id="ABC"
              />,
              <DeleteItem
                id="ABC"
              >
                Delete This Item
              </DeleteItem>,
            ],
            "className": "buttonList",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <a>
                  Edit ✏️
                </a>,
                "href": Object {
                  "pathname": "update",
                  "query": Object {
                    "id": "ABC",
                  },
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Edit ✏️",
                },
                "ref": null,
                "rendered": "Edit ✏️",
                "type": "a",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "id": "ABC",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Delete This Item",
                "id": "ABC",
              },
              "ref": null,
              "rendered": "Delete This Item",
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
